### PR TITLE
Implement pull request previews using Cloudflare

### PR DESCRIPTION
**Pull request by zregvart**
This implements publishing to preview sites on Cloudflare pages by utilizing the build reusable workflow to generate the site artifacts and uploading them using wrangler, managed by `package.json`. When the preview workflow succeeds, a comment is placed on the pull request pointing to the published URL.

Also includes: [Use reusable workflow for build](https://github.com/enterprise-contract/enterprise-contract.github.io/commit/1396e4a9fb34ac01bdf955e17bbde07a6104e6e7)
This way we can reuse this workflow in publishing from main and from pull requests.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,104 @@
+# Copyright 2022 Red Hat, Inc.
```
Comment (lcarva) [inline .github/workflows/build.yaml]: Update copyright year?

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        uses: actions/setup-go@v4
+        with:
+          go-version-file: antora/go.mod
+          cache: true
+          cache-dependency-path: antora/go.sum
+
+      - name: Generate website
+        run: npm --prefix antora ci && npm --prefix antora run build
+
+      - name: Upload artifact
```
Comment (lcarva) [inline .github/workflows/build.yaml:50]: I see this workflow can be triggered on pull requests. Will that cause any issues when multiple PRs are uploading artifacts? Not sure if they're properly scoped to the PR. Maybe the `path` below need to have the PR number?

UPDATE: The `preview` workflow seems to indicate that artifacts are properly scoped to a specific run. :tada:

**Review by lcarva — APPROVED** [review]

**Review by simonbaird — APPROVED** [review]
Let's merge and see if it works. :cowboy_hat_face: .

**Comment by simonbaird** [comment]
We can see how it goes with #42.

File: .github/workflows/build.yaml (hunk 2 above)
Comment (zregvart) [reply .github/workflows/build.yaml:50]: Artifacts are property of workflow runs

**Comment by zregvart** [comment]
There was a typo in the workflow so I pushed https://github.com/enterprise-contract/enterprise-contract.github.io/commit/ec9fdb24c7e878c22d0a7eba2af50c0220975411 directly

**Pull request merged**